### PR TITLE
SEARCH-840 Migrates SwiftSearch to a different project

### PR DESCRIPTION
**Pull request by keerthi16**
## Description
Migrates SwiftSearch to a different project and including that as an `optionalDependencies` in SymphonyElectron
[SEARCH-840](https://perzoinc.atlassian.net/browse/SEARCH-840)

_Note_
- Modified DEV_SETUP.md
- Removes `FFI`, `electron-utils` and `ref` from dependencies
- Removes Search unit-tests
- Removes Search Demo page

## Solution Approach
Removes all the code related to SwiftSearch and now its been added as a plugin(optionalDependencies).

## Documentation
Here is the link to the project [SwiftSearch](https://github.com/symphonyoss/SwiftSearch)
Which is migrated from Javascript to typescript.

## QA Checklist
- [x] Unit-Tests
[Test Results — Unit-Tests.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2208282/Test.Results.Unit-Tests.pdf)
- [x] Automation-Tests
[Test Results — Spectron.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2208283/Test.Results.Spectron.pdf)

### CODE REVIEW

File: js/preload/preloadMain.js [hunk 1]
```diff
@@ -28,6 +28,23 @@ const KeyCodes = {
...
 };
 
+let Search;
+let SearchUtils;
+
+try {
+    Search = remote.require('swift-search').Search;
+} catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn("Failed to initializing Search");
```
Comment (VishwasShashidhar) [inline js/preload/preloadMain.js]: Please change this to `Failed to initialize swift search. You'll need to include the search dependency. Contact the developers for more details`

File: js/preload/preloadMain.js [hunk 2]
```diff
@@ -28,6 +28,23 @@ const KeyCodes = {
...
+}
+
+try {
+    SearchUtils = remote.require('swift-search').SearchUtils;
+} catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn("Failed to initializing SearchUtils");
```
Comment (VishwasShashidhar) [inline js/preload/preloadMain.js]: Please change this to `Failed to initialize swift search. You'll need to include the search dependency. Contact the developers for more details`

File: package.json [hunk 3]
```diff
@@ -50,10 +48,7 @@
     "extraFiles": [
       "config/Symphony.config",
       "library/libsymphonysearch.dylib",
-      "library/indexvalidator.exec",
-      "library/lz4.exec",
-      "library/search-launch-agent.sh",
-      "library/search-launch-daemon.sh"
+      "library/lz4.exec"
```
Comment (VishwasShashidhar) [inline package.json]: is this still required? if so, what is it for?

Comment (keerthi16) [reply package.json]: Yes, When we build the electron we need to copy these libraries.

**Pull request merged**